app/assets/javascripts/discourse/app/mixins/textarea-text-manipulation.js

```diff
@@ -526,6 +526,10 @@ export default Mixin.create({
       return;
     }
 
+    if (this.isInsideCodeFence(text, offset)) {
```

Comment:
Could you use `isInside` like on line 398?

```
const isCodeBlock = this.isInside(pre, /(^|\n)```/g);
```

Reply:
god xD I looked for a similar codepath and didnt see it, thanks